### PR TITLE
use tc mirred rule to connect hypervisor nic and container nic

### PR DESCRIPTION
**Pull request by gao-feng**
Try to fix https://github.com/hyperhq/runv/issues/582

this pr adds tc mirred rules in the netns of shim process to connect two type of nics together without regarding for the docker network mode.

macvtap may be an alternate, but it can only be used on qemu(xen?). 

need more test.

### CODE REVIEW

File: lib/runvxenlight/xenlight-runv.go [hunk 1]
```diff
@@ -124,14 +124,15 @@ void runvxl_childproc_setmode(libxl_ctx *ctx) {
...
-int runvxl_add_nic(libxl_ctx *ctx, int32_t domid, int id, char *bridge, char *mac) {
+int runvxl_add_nic(libxl_ctx *ctx, int32_t domid, int id, char *bridge, char *device, char *mac) {
     libxl_device_nic nic;
     int ret;
 
     libxl_device_nic_init(&nic);
     nic.bridge = strdup(bridge);
     nic.nictype = LIBXL_NIC_TYPE_VIF;
     nic.devid = id;
+    nic.ifname = strdup(device);
```
Comment (laijs) [inline lib/runvxenlight/xenlight-runv.go]: where is the code that nic.ifname and nic.bridge are to be freed?
Is it libxl_device_nic_dispose(&nic); ?

File: cli/network.go [hunk 2]
```diff
@@ -53,68 +53,14 @@ type nsListener struct {
...
+func createFakeBridge() {
+	// add an useless bridge to satisfy hypervisor, most of them need to join bridge.
+	la := netlink.NewLinkAttrs()
+	la.Name = fakeBridge
+	bridge := &netlink.Bridge{LinkAttrs: la}
+	if err := netlink.LinkAdd(bridge); err != nil && !os.IsExist(err) {
+		glog.Errorf("fail to create fake bridge: %v, %v", fakeBridge, err)
 	}
-
-	if bridge.Name == "ovs-system" {
```
Comment (laijs) [inline cli/network.go]: Why is this part removed? Is this code also a trick?

File: cli/network.go [hunk 3]
```diff
@@ -25,6 +24,7 @@ const (
 	UpdateTypeLink  NetlinkUpdateType = "link"
 	UpdateTypeAddr  NetlinkUpdateType = "addr"
 	UpdateTypeRoute NetlinkUpdateType = "route"
+	fakeBridge      string            = "runv0"
```
Comment (laijs) [inline cli/network.go]: all taps will be added to this bridge. I'm afraid it might cause information leak.

Comment (laijs) [reply cli/network.go]: when the taps are moved into shim's ns, they are also atomically moved out from the bridge. But what happen before the moving?

File: lib/runvxenlight/xenlight-runv.go (hunk 1 above)
Comment (gao-feng) [reply lib/runvxenlight/xenlight-runv.go]: yes

File: cli/network.go (hunk 2 above)
Comment (gao-feng) [reply cli/network.go]: Nop, the new design doesn't need to lookup bridge/ovs on host now

File: cli/network.go (hunk 3 above)
Comment (gao-feng) [reply cli/network.go]: Have no idea, What's your opinion?

**Comment by laijs** [comment]
LGTM
